### ISSUE TITLE
dataconnect @auth directive - role based auth expression with tokens doesnt work.

### ISSUE THREAD

**Issue by artanisdesign** [firebase/firebase-tools#7944]
### Environment info

**firebase-tools: 13.25.0**

**Platform: mac **

###  Test case

` mutation createItem(
   $title: String!
 ) @auth(expr: "auth.token.isAdmin == true") {
    item_insert(
      data: {
        title: $title
      }
    )
 }`

### Steps to reproduce

I run this mutation locally from data connect execution, where i do set authentication and variables too. 

### Expected behavior

add item only when my role is admin. 

### Actual behavior

weird error:
```
mutation, createItem On @auth: invalid expression ("auth.token.isAdmin == true"): ERROR: <input>:1:5: unexpected failed resolution of 'celenv.AuthUserInfo'
 | auth.token.isAdmin == true
 | ....^
```
that only happens locally. but because of this error i can not deploy to production.

**Comment by artanisdesign** [comment]
hi @aalej  yea, i noticed that part - it only occurs locally-, its in the original issue too. while from admin or console i can do it, i cant deploy the named mutation to do console.. so obviously i cant use that from my client code.

**Comment by joehan** [comment]
Hey @artanisdesign - thanks for reporting this and apologies for the inconvenience. This should be fixed by https://github.com/firebase/firebase-tools/pull/7968, which will go out in the next release this afternoon.

**Comment by artanisdesign** [comment]
Great, thanks!On 2024. Nov 19., at 19:24, joehan ***@***.***> wrote:﻿
Hey @artanisdesign - thanks for reporting this and apologies for the inconvenience. This should be fixed by #7968, which will go out in the next release this afternoon.
